### PR TITLE
[CORE] Various fixes/improvements for unit test

### PR DESCRIPTION
**Pull request by smuzaffar**
Updated unit tests so that they can be run from their own separate directory. Mostly
- convert <bin ...> to <test....>
- avoid creating test output in cmssw/tmp
- added flag `USE_UNITTEST_DIR="1"` to run each tests in separate dir. Once all the tests can run it their own tmp path then we can enable  `USE_UNITTEST_DIR="1"` at project level and do cleanup

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

File: IOPool/Streamer/test/RunSimple_NewStreamer.sh [hunk 1]
```diff
@@ -2,43 +2,29 @@
...
-cmsRun --parameter-set NewStreamInAlt_cfg.py  > alt  2>&1 || die "cmsRun NewStreamInAlt_cfg.py" $?
-cmsRun --parameter-set NewStreamInExt_cfg.py  > ext  2>&1 || die "cmsRun NewStreamInExt_cfg.py" $?
-cmsRun --parameter-set NewStreamInExtBuf_cfg.py  > ext  2>&1 || die "cmsRun NewStreamInExtBuf_cfg.py" $?
+cmsRun ${SCRAM_TEST_PATH}/NewStreamOut_cfg.py compAlgo=${TEST_COMPRESSION_ALGO} > out 2>&1 || die "cmsRun NewStreamOut_cfg.py compAlgo=${TEST_COMPRESSION_ALGO}" $?
+cmsRun ${SCRAM_TEST_PATH}/NewStreamOutAlt_cfg.py compAlgo=${TEST_COMPRESSION_ALGO} > outAlt 2>&1 || die "cmsRun NewStreamOut_cfg.py compAlgo=${TEST_COMPRESSION_ALGO}" $?
+cmsRun ${SCRAM_TEST_PATH}/NewStreamOutExt_cfg.py compAlgo=${TEST_COMPRESSION_ALGO} > outExt 2>&1 || die "cmsRun NewStreamOut_cfg.py compAlgo=${TEST_COMPRESSION_ALGO}" $?
+cmsRun ${SCRAM_TEST_PATH}/NewStreamOutExt2_cfg.py compAlgo=${TEST_COMPRESSION_ALGO} > outExt 2>&1 || die "cmsRun NewStreamOutExt2_cfg.py compAlgo=${TEST_COMPRESSION_ALGO}" $?
+cmsRun --parameter-set ${SCRAM_TEST_PATH}/NewStreamIn_cfg.py  > in  2>&1 || die "cmsRun NewStreamIn_cfg.py" $?
```
[the remark this answers is not in the document]
Comment (smuzaffar) [reply IOPool/Streamer/test/RunSimple_NewStreamer.sh]: done

File: IOPool/Common/test/TestEdmFastMerge.sh [hunk 2]
```diff
@@ -2,74 +2,71 @@
...
 egrep -v "<GUID>|<PFN>|^$" $LOCAL_TMP_DIR/TestFastMergeFJRx.xml  > $LOCAL_TMP_DIR/TestFastMergeFJRx_filtered.xml
 diff $LOCAL_TMP_DIR/proper_fjrx_output_filtered $LOCAL_TMP_DIR/TestFastMergeFJRx_filtered.xml || die 'output framework job report is wrong for proper_fjrx_output_filtered' $?
 
-#cmsRun -j ${LOCAL_TMP_DIR}/TestFastMergeFJRx_second.xml --parameter-set ${LOCAL_TEST_DIR}/FastMergeTest_x_second_cfg.py || die 'Failure using FastMergeTest_x_second_cfg.py' $?
+#cmsRun -j ${LOCAL_TMP_DIR}/TestFastMergeFJRx_second.xml ${LOCAL_TEST_DIR}/FastMergeTest_x_second_cfg.py || die 'Failure using FastMergeTest_x_second_cfg.py' $?
 ##need to filter items in job report which always change
 #egrep -v "<GUID>|<PFN>|^$" $LOCAL_TEST_DIR/proper_fjrx_second_output > $LOCAL_TMP_DIR/proper_fjrx_second_output_filtered
 #egrep -v "<GUID>|<PFN>|^$" $LOCAL_TMP_DIR/TestFastMergeFJRx_second.xml  > $LOCAL_TMP_DIR/TestFastMergeFJRx_second_filtered.xml
 #diff $LOCAL_TMP_DIR/proper_fjrx_second_output_filtered $LOCAL_TMP_DIR/TestFastMergeFJRx_second_filtered.xml || die 'output framework job report is wrong for proper_fjrx_second_output_filtered' $?
```
Comment (makortel) [inline IOPool/Common/test/TestEdmFastMerge.sh:51]: I wonder if this commented out test should be enabled (possibly in a later PR), or removed. I see it was added (as commented out) in 5f1487b85d0e.

Comment (smuzaffar) [reply IOPool/Common/test/TestEdmFastMerge.sh:51]: @Dr15Jones what do you suggest here? 
@makortel , changes for this can go in a separate PR

**Comment by perrotta** [comment]
+1

**Pull request merged**